kairon/shared/data/action_serializer.py

```
@@ -0,0 +1,467 @@
+                                action_names.append(action_name)
+                                new_actions.append(a)
+                        filtered_actions[action_type] = new_actions
+
+            for action_type, data in filtered_actions.items():
+                if data:
+                    ActionSerializer.save_collection_data_list(action_type, bot, user, data)
+        if other_collections_data:
+            ActionSerializer.save_other_collections(other_collections_data, bot, user, overwrite)
+
+    @staticmethod
+    def get_action_config_data_list(bot: str, action_model: Document, with_doc_id: bool = False, query: dict = {}) -> list[dict]:
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)